core/src/main/java/io/questdb/std/datetime/DateLocaleFactory.java

```diff
@@ -52,6 +56,18 @@ public static void load() {
     }
 
     public DateLocale getLocale(CharSequence id) {
-        return dateLocales.get(id);
+        DateLocale dateLocale = dateLocales.get(id);
+        if (dateLocale == null) {
+            return null;
+        }
+        if (dateLocale != dummyLocale) {
+            return dateLocale;
+        }
+        return dateLocales.compute(id, computeDateLocaleBiFunc);
+    }
+
+    private DateLocale computeDateLocale(CharSequence key, DateLocale val) {
```

Comment:
I'd add this fast path check to avoid wasted CPU time and allocations under concurrent `computeDateLocale` calls:
```java
        if (val != dummyLocale) {
            // Someone was faster than us.
            return val;
        }
```

Reply:
Added.